nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentImpl.java

```diff
@@ -20,136 +20,243 @@
 
 package org.nuxeo.ecm.platform.comment.api;
 
+import static org.nuxeo.common.utils.DateUtils.toCalendar;
+import static org.nuxeo.common.utils.DateUtils.toInstant;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_ANCESTOR_IDS_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_AUTHOR_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_CREATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_DOC_TYPE;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_MODIFICATION_DATE_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_PARENT_ID_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.CommentConstants.COMMENT_TEXT_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.ExternalEntityConstants.EXTERNAL_ENTITY_FACET;
+import static org.nuxeo.ecm.platform.comment.api.ExternalEntityConstants.EXTERNAL_ENTITY_ID_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.ExternalEntityConstants.EXTERNAL_ENTITY_ORIGIN_PROPERTY;
+import static org.nuxeo.ecm.platform.comment.api.ExternalEntityConstants.EXTERNAL_ENTITY_PROPERTY;
+
 import java.time.Instant;
+import java.util.Arrays;
+import java.util.Calendar;
 import java.util.Collection;
 import java.util.HashSet;
-import java.util.Objects;
 
 import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.impl.SimpleDocumentModel;
 
 /**
  * @since 10.3
  */
 public class CommentImpl implements Comment, ExternalEntity {
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String id;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String parentId;
 
```

Comment:
is this property will be removed in the future, my question is related to fact that in some commit (more in the cleanup one) you use `@Deprecated(since = "11.1", forRemoval = true)`

Reply:
IMO, everything we deprecate is to be removed, `forRemoval` or not.

Reply:
Indeed, will remove the introduced `forRemoval`.

We will soon try to rewrite coding style in order to take into account such things.